core/src/main/kotlin/com/imashnake/animite/core/ui/Icon.kt

```diff
@@ -0,0 +1,95 @@
+package com.imashnake.animite.core.ui
+
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.rounded.AccountCircle
+import androidx.compose.material.icons.rounded.Face
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.IconButtonDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.imashnake.animite.core.R
+
+/**
+ * Wrapper for [IconButton]. Flattens an [IconButton] with an [Icon].
+ *
+ * @param onClick called when this icon button is clicked.
+ * @param imageVector [ImageVector] to draw inside this icon.
+ * @param modifier the [Modifier] to be applied to this icon button.
+ * @param foregroundColor the color of the [Icon].
+ * @param backgroundColor the background color of the [IconButton].
+ * @param contentDescription text used by accessibility services to describe what this icon.
+ */
+@Composable
+fun IconButton(
+    onClick: () -> Unit,
+    imageVector: ImageVector,
+    modifier: Modifier = Modifier,
+    foregroundColor: Color = MaterialTheme.colorScheme.onPrimary,
+    backgroundColor: Color = MaterialTheme.colorScheme.primary,
+    contentDescription: String? = null
+) {
+    IconButton(
+        onClick = onClick,
+        modifier = modifier.size(dimensionResource(R.dimen.icon_size)),
+        colors = IconButtonDefaults.iconButtonColors(
+            containerColor = backgroundColor,
+            contentColor = foregroundColor
+        )
+    ) {
+        Icon(
+            imageVector = imageVector,
+            contentDescription = contentDescription,
+            tint = foregroundColor
+        )
+    }
+}
+
+@Composable
+fun Icon(
+    imageVector: ImageVector,
+    modifier: Modifier = Modifier,
+    tint: Color = MaterialTheme.colorScheme.onPrimary,
+    contentDescription: String? = null
+) {
+    Icon(
+        imageVector = imageVector,
+        contentDescription = contentDescription,
+        modifier = modifier
+            .size(dimensionResource(R.dimen.icon_size))
```

Comment:
What's wrong with the default icon size?

Reply:
My intention was to have a component that we can reuse but I suppose that can happen when it has to.

Reply:
The icon didn't snap to the grid in Figma and was imported as 24 x 23 instead of 24 x 24 so we had an oval search bar, I (incorrectly) enforced this by using `Modifier.size()`. Fixed the drawable and removed `size` :bread: